api/v1beta1/foundationdbcluster_types.go

```diff
@@ -94,6 +94,11 @@ type FoundationDBClusterSpec struct {
 	// After the initial reconciliation, this will not be used.
 	SeedConnectionString string `json:"seedConnectionString,omitempty"`
 
+	// PartialConnectionString provides a way to specify part of the connection string
+	// (e.g. the database name and coordinator generation) without specifying the entire
+	// string.
+	PartialConnectionString ConnectionString `json:"partialConnectionString,omitempty"`
```

Comment:
Am I right in thinking that this cannot be used to specify the coordinators? If so, it may be good to make that explicit in the documentation.

Reply:
Done.

Reply:
I wonder if we should merge that with `SeedConnectionString` otherwise we have multiple different settings to change the behaviour e.g. adding a bool `onlyForInitialize` and then. deprecate `SeedConnectionString`?

Reply:
We wouldn’t need the bool as the semantics of both are only for initialization. The reason to keep both is that forcing users to decompose a connection string they already have into its constituent parts is rather annoying. I see a case for both ways and am happy to go with whichever option the two of you would prefer.

Reply:
I think they're going to solve different purposes, and there's a good chance their usage will diverge in the long-terms. I see `seedConnectionString` as a hack for bootstrapping multi-KC clusters, and we want to develop a more robust solution that also allows instances of the operator to discover new connection strings from their peers if they get partitioned.

Reply:
That sounds reasonable. My only concern is/was that we add to many different settings that solve a similar problem. In the end it should be easy for an user to know which settings are required for a specific task.

Reply:
One NIT: wee may want to document if both `seedConnectionString` and `partialConnectionString` are set `seedConnectionString` will be used and `partialConnectionString` will be ignored (?) and the TLDR would be don't use both :)